js_modules/dagit/packages/core/src/hooks/useStateWithStorage.tsx

```diff
@@ -38,10 +55,15 @@ export function useStateWithStorage<T>(key: string, validate: (json: any) => T)
       } else {
         window.localStorage.setItem(key, JSON.stringify(next));
       }
+      document.removeEventListener(DID_WRITE_LOCALSTORAGE, listener);
+      document.dispatchEvent(new CustomEvent(DID_WRITE_LOCALSTORAGE, {detail: key}));
+      document.addEventListener(DID_WRITE_LOCALSTORAGE, listener);
```

Comment:
What's going on here?

Reply:
Ahh this is a little subtle - basically if you have two components on any page that use `useStateWithStorage` for the SAME state key, they do not stay in sync. eg: you can toggle the view mode, and the other component reading view mode via useStateWithStorage will not see the new value.

This broadcasts a change event as a custom DOM event that allows other hooks on the page reading the same localstorage key to update and emit the new value.

I think we could alternatively implement it as a global instance of a new class that inherits from EventEmitter, etc., but this was the most compact solution I could think of.